services/block_service.go

```diff
@@ -68,5 +68,27 @@ func (s *BlockAPIService) BlockTransaction(
 	ctx context.Context,
 	request *types.BlockTransactionRequest,
 ) (*types.BlockTransactionResponse, *types.Error) {
-	return nil, wrapErr(ErrUnimplemented, nil)
+	if s.config.Mode != configuration.Online {
+		return nil, ErrUnavailableOffline
+	}
+
+	if request.BlockIdentifier == nil || request.TransactionIdentifier == nil {
+		return nil, wrapErr(ErrInvalidInput, errors.New("block_identifier or transaction_identifier cannot be empty"))
+	}
+	if request.BlockIdentifier.Hash == "" && request.BlockIdentifier.Index != 0{
+		return nil, wrapErr(ErrInvalidInput, errors.New("at least on of block hash or index should be present"))
+	}
+
+	if request.TransactionIdentifier.Hash == "" {
+		return nil, wrapErr(ErrInvalidInput, errors.New("transaction hash is not provided"))
+	}
```

Comment:
All the block identifier / hash / tx identifier checks are not needed, since we are using `rosetta-sdk-go`'s `BlockAPIController`, and the API controller already checks these.
By the time we enter to `block_service`, those values are already checked.

See:
https://github.com/coinbase/rosetta-sdk-go/blob/master/server/api_block.go#L106